### ISSUE TITLE
508-defect-2 [SCREENREADER]: The text reminders checkbox label should be descriptive

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23169]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect the text reminders checkbox label to be descriptive so I don't confuse it as being connected to my work number.

## Details

Some screen reader users may misinterpret the text reminder checkbox as being associated to their work number as opposed to their mobile number due to a combination of the tab order and lack of description within the checkbox label.

For example, upon tabbing to the work phone edit button, they will hear "Edit work phone", upon tabbing again, they will hear "We'll send VA health care appointment text reminders to this number" and may assume "this number" is referring to their work phone and not their mobile number.

## Acceptance Criteria

- [ ] Text reminders checkbox label is descriptive

## Steps to Recreate

My profile information page, user 36 Wesley, upon tabbing to the text reminders checkbox

## Proposed Solution (if known)

Instead of "this number," use "your mobile number" in the checkbox label.

e.g. We'll send VA health care appointment text reminders to ~~this~~ your mobile number 

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/114617859-a7c96600-9c76-11eb-9bec-185b12d6e420.mov

**Comment by erikphansen** [comment]
This can be validated on staging. But do it quick because we are also about to totally change up this UI!

**Comment by joshkimux** [comment]
<img width="1029" alt="Screen Shot 2021-05-14 at 12 02 54 PM" src="https://user-images.githubusercontent.com/14154792/118297555-521dee80-b4ac-11eb-9ecb-2f99dd0a0aab.png">

@erikphansen 🔥 🔥 🔥
